File: grass-next-prj/GrassNext_project/GrassNextFrontend/src/app/model/gps-point.ts
/**
 * Class representing a geographical point using GPS coordinates.
 */
export class GpsPoint {
    /**
     * Constructs a new instance with the specified latitude and longitude.
     *
     * @param {number} latitude - The geographic latitude.
     * @param {number} longitude - The geographic longitude.
     *
     */
    constructor(
      public latitude: number,
      public longitude: number
    ) {}
}

File: grass-next-prj/GrassNext_project/GrassNextServer/src/main/java/com/grassnext/grassnextserver/util/gpspoint/GpsPoint.java
package com.grassnext.grassnextserver.util.gpspoint;

import jakarta.persistence.Embeddable;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Represents a GPS point with latitude and longitude coordinates. This class is embeddable
 * in other entities allowing for the storage of geographic location coordinates.
 *
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@Embeddable
public class GpsPoint {
    /**
     * Represents the longitude coordinate of a GPS point.
     *
     */
    private Double longitude;
    /**
     * Represents the latitude coordinate of a GPS point.
     *
     */
    private Double latitude;
}